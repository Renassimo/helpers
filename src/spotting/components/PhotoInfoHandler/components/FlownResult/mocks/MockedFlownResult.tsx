const MockedFlownResult = jest.fn(() => <div>MockedFlownResult</div>);

export default MockedFlownResult;
