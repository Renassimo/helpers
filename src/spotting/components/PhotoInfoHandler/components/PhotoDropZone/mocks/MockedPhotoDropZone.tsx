const MockedPhotoDropZone = jest.fn(() => <div>MockedPhotoDropZone</div>);

export default MockedPhotoDropZone;
