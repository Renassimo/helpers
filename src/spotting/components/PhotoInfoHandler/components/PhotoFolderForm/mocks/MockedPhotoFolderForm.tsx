const MockedPhotoFolderForm = jest.fn(() => <div> MockedPhotoFolderForm</div>);

export default MockedPhotoFolderForm;
