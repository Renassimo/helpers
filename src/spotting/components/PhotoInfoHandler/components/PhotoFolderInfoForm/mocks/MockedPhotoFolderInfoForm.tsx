const MockedPhotoFolderInfoForm = jest.fn(() => (
  <div>MockedPhotoFolderInfoForm</div>
));

export default MockedPhotoFolderInfoForm;
