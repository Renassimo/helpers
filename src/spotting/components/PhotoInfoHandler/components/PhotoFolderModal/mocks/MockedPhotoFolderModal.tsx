const MockedPhotoFolderModal = jest.fn(() => <div>MockedPhotoFolderModal</div>);

export default MockedPhotoFolderModal;
