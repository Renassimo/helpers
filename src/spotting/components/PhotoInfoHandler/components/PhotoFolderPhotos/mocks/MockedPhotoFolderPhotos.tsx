const MockedPhotoFolderPhotos = jest.fn(() => (
  <div>MockedPhotoFolderPhotos</div>
));

export default MockedPhotoFolderPhotos;
