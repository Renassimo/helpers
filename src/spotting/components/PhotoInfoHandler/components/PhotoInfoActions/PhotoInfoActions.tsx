import Box from '@mui/material/Box';
import Button from '@mui/material/Button';
import Typography from '@mui/material/Typography';

import { PhotoActionType } from '@/spotting/types';

import usePhotoInfoContext from '@/spotting/contexts/hooks/usePhotoInfoContext';

const PhotoInfoActions = () => {
  const { photosList, handlingText, dispatch } = usePhotoInfoContext();

  const clearSelections = () =>
    dispatch({ type: PhotoActionType.UNSELECT_ALL });
  const selectAll = () => dispatch({ type: PhotoActionType.SELECT_ALL });
  const deleteAll = () => dispatch({ type: PhotoActionType.CLEAR_FILES });
  const createFolder = () => dispatch({ type: PhotoActionType.CREATE_FOLDER });

  return (
    <Box>
      {photosList.length > 0 && (
        <>
          <Button onClick={selectAll}>Select all</Button>
          <Button onClick={clearSelections}>Clear selections</Button>
          <Button onClick={deleteAll}>Delete all</Button>
          <Button onClick={createFolder}>Create folder</Button>
        </>
      )}
      {handlingText && <Typography>{handlingText}</Typography>}
    </Box>
  );
};

export default PhotoInfoActions;
