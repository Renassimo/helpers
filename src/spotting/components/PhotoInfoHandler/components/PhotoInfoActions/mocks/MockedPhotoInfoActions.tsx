const MockedPhotoInfoActions = jest.fn(() => <div>MockedPhotoInfoActions</div>);

export default MockedPhotoInfoActions;
