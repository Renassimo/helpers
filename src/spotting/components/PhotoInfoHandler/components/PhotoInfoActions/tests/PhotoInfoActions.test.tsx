import userEvent from '@testing-library/user-event';
import renderWithTheme from '@/common/tests/helpers/renderWithTheme';

import usePhotoInfoContext from '@/spotting/contexts/hooks/usePhotoInfoContext';

import { mockedPhotosList } from '@/spotting/types/mocks';

import { PhotoActionType } from '@/spotting/types';

import PhotoInfoActions from '../PhotoInfoActions';

jest.mock('@/spotting/contexts/hooks/usePhotoInfoContext');

describe('PhotoInfoActions', () => {
  const dispatch = jest.fn();

  afterEach(() => {
    jest.clearAllMocks();
  });

  test('renders successfully', () => {
    // Arange
    (usePhotoInfoContext as unknown as jest.Mock).mockImplementation(
      jest.fn(() => ({
        photosList: [],
        handlingText: 'handling-text',
        dispatch,
      }))
    );
    // Act
    const { baseElement } = renderWithTheme(<PhotoInfoActions />);
    // Assert
    expect(baseElement).toMatchSnapshot();
  });

  describe('when photosList passed', () => {
    test('renders successfully', () => {
      // Arange
      (usePhotoInfoContext as unknown as jest.Mock).mockImplementation(
        jest.fn(() => ({
          photosList: mockedPhotosList,
          handlingText: 'handling-text',
          dispatch,
        }))
      );
      // Act
      const { baseElement } = renderWithTheme(<PhotoInfoActions />);
      // Assert
      expect(baseElement).toMatchSnapshot();
    });

    describe('when clicks Select all', () => {
      test('calls dispatch', async () => {
        // Arange
        (usePhotoInfoContext as unknown as jest.Mock).mockImplementation(
          jest.fn(() => ({
            photosList: mockedPhotosList,
            handlingText: 'handling-text',
            dispatch,
          }))
        );
        const { getByText } = renderWithTheme(<PhotoInfoActions />);
        // Act
        await userEvent.click(getByText('Select all'));
        // Assert
        expect(dispatch).toBeCalledWith({ type: PhotoActionType.SELECT_ALL });
      });
    });

    describe('when clicks Clear selections', () => {
      test('calls dispatch', async () => {
        // Arange
        (usePhotoInfoContext as unknown as jest.Mock).mockImplementation(
          jest.fn(() => ({
            photosList: mockedPhotosList,
            handlingText: 'handling-text',
            dispatch,
          }))
        );
        const { getByText } = renderWithTheme(<PhotoInfoActions />);
        // Act
        await userEvent.click(getByText('Clear selections'));
        // Assert
        expect(dispatch).toBeCalledWith({ type: PhotoActionType.UNSELECT_ALL });
      });
    });

    describe('when clicks Delete all', () => {
      test('calls dispatch', async () => {
        // Arange
        (usePhotoInfoContext as unknown as jest.Mock).mockImplementation(
          jest.fn(() => ({
            photosList: mockedPhotosList,
            handlingText: 'handling-text',
            dispatch,
          }))
        );
        const { getByText } = renderWithTheme(<PhotoInfoActions />);
        // Act
        await userEvent.click(getByText('Delete all'));
        // Assert
        expect(dispatch).toBeCalledWith({
          type: PhotoActionType.CLEAR_FILES,
        });
      });
    });

    describe('when clicks Create folder', () => {
      test('calls dispatch', async () => {
        // Arange
        (usePhotoInfoContext as unknown as jest.Mock).mockImplementation(
          jest.fn(() => ({
            photosList: mockedPhotosList,
            handlingText: 'handling-text',
            dispatch,
          }))
        );
        const { getByText } = renderWithTheme(<PhotoInfoActions />);
        // Act
        await userEvent.click(getByText('Create folder'));
        // Assert
        expect(dispatch).toBeCalledWith({
          type: PhotoActionType.CREATE_FOLDER,
        });
      });
    });
  });
});
