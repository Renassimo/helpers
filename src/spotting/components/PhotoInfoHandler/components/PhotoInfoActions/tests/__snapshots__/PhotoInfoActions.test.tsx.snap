// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PhotoInfoActions renders successfully 1`] = `
<body>
  <div>
    <div
      class="MuiBox-root css-0"
    >
      <p
        class="MuiTypography-root MuiTypography-body1 css-ahj2mt-MuiTypography-root"
      >
        handling-text
      </p>
    </div>
  </div>
</body>
`;

exports[`PhotoInfoActions when photosList passed renders successfully 1`] = `
<body>
  <div>
    <div
      class="MuiBox-root css-0"
    >
      <button
        class="MuiButtonBase-root MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium MuiButton-colorPrimary MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium MuiButton-colorPrimary css-1e6y48t-MuiButtonBase-root-MuiButton-root"
        tabindex="0"
        type="button"
      >
        Select all
        <span
          class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
        />
      </button>
      <button
        class="MuiButtonBase-root MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium MuiButton-colorPrimary MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium MuiButton-colorPrimary css-1e6y48t-MuiButtonBase-root-MuiButton-root"
        tabindex="0"
        type="button"
      >
        Clear selections
        <span
          class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
        />
      </button>
      <button
        class="MuiButtonBase-root MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium MuiButton-colorPrimary MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium MuiButton-colorPrimary css-1e6y48t-MuiButtonBase-root-MuiButton-root"
        tabindex="0"
        type="button"
      >
        Delete all
        <span
          class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
        />
      </button>
      <button
        class="MuiButtonBase-root MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium MuiButton-colorPrimary MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium MuiButton-colorPrimary css-1e6y48t-MuiButtonBase-root-MuiButton-root"
        tabindex="0"
        type="button"
      >
        Create folder
        <span
          class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
        />
      </button>
      <p
        class="MuiTypography-root MuiTypography-body1 css-ahj2mt-MuiTypography-root"
      >
        handling-text
      </p>
    </div>
  </div>
</body>
`;
